### PR TITLE
Support overwriting probes when the port is a named port.

### PR DESCRIPTION
**Pull request by ishustava**
Previously, when transparent proxy is enabled and overwrite probes is true,
we would assume that the port value of an HTTP probe is an integer, however,
it can also be a name that refers to the named container port.
This change makes it possible to overwrite probes that refer to named container
ports.

How I've tested this PR:
unit tests

How I expect reviewers to test this PR:
code review

Checklist:
- [x] Tests added
- [x] CHANGELOG entry added (*HashiCorp engineers only, community PRs should not add a changelog entry*)

### CODE REVIEW

**Comment by kschoche** [comment]
Just passing through, but really excellent work, I love the cleanup pieces as well, it helps the mutation portion read more easily!!!! 👍🏽

**Review by lkysow — APPROVED** [review]
👽

File: connect-inject/endpoints_controller.go [hunk 1]
```diff
@@ -532,28 +532,19 @@ func (r *EndpointsController) createServiceRegistrations(pod corev1.Pod, service
...
-				} else if sp.TargetPort.Type == intstr.Int && sp.TargetPort.IntVal != 0 {
-					// If the target port is a non-zero int, we can compare that port directly with the Consul service port.
-					if sp.TargetPort.IntVal == int32(consulServicePort) {
-						k8sServicePort = sp.Port
-						break
-					}
+				// If the targetPortValue is not zero and is the consulServicePort, then this is the service port we'll use as the tagged address.
+				if targetPortValue != 0 && targetPortValue == consulServicePort {
+					k8sServicePort = sp.Port
+					break
```
Comment (thisisnotashwin) [inline connect-inject/endpoints_controller.go:547]: Love this cleanup!!

File: connect-inject/endpoints_controller_test.go [hunk 2]
```diff
@@ -4215,6 +4215,85 @@ func TestCreateServiceRegistrations_withTransparentProxy(t *testing.T) {
 			expExposePaths: nil,
 			expErr:         "",
 		},
+		"probes with port names": {
+			tproxyGlobalEnabled: true,
+			overwriteProbes:     true,
+			podAnnotations: map[string]string{
+				annotationOriginalPod: "{\"metadata\":{\"name\":\"test-pod-1\",\"namespace\":\"default\",\"creationTimestamp\":null,\"labels\":{\"consul.hashicorp.com/connect-inject-managed-by\":\"consul-k8s-endpoints-controller\",\"consul.hashicorp.com/connect-inject-status\":\"injected\"}},\"spec\":{\"containers\":[{\"name\":\"test\",\"ports\":[{\"name\":\"tcp\",\"containerPort\":8081},{\"name\":\"http\",\"containerPort\":8080}],\"resources\":{},\"livenessProbe\":{\"httpGet\":{\"port\":\"tcp\"}},\"readinessProbe\":{\"httpGet\":{\"port\":\"http\"}},\"startupProbe\":{\"httpGet\":{\"port\":\"http\"}}}]},\"status\":{\"hostIP\":\"127.0.0.1\",\"podIP\":\"1.2.3.4\"}}\n",
```
Comment (thisisnotashwin) [inline connect-inject/endpoints_controller_test.go:4222]: :chefkiss:

**Review by thisisnotashwin — COMMENTED** [review]
🦙 this is very excellent!!

**Review by thisisnotashwin — APPROVED** [review]

**Pull request merged**